saleor/payment/utils.py

```diff
@@ -189,12 +212,91 @@ def create_payment_information(
         reuse_source=store_source,
         data=additional_data or {},
         graphql_customer_id=graphql_customer_id,
+        refund_data=refund_data,
         _resolve_lines=lambda: create_payment_lines_information(
             payment, manager or get_plugins_manager()
         ),
     )
 
 
+RefundLines = Iterator[Tuple[Any, OrderLine]]
+
+
+def _prepare_refund_lines(
+    fulfillments: QuerySet[Fulfillment],
+    order_lines_to_refund: List[OrderLineData],
+    fulfillment_lines_to_refund: List[FulfillmentLineData],
+) -> Iterator[Tuple[int, int]]:
+    previous_refund_lines = chain.from_iterable(
+        (
+            (p_variant_id, line.quantity)
+            for line in f.lines.filter(order_line__variant_id__isnull=False)
+            if (p_variant_id := line.order_line.variant_id)
+        )
+        for f in fulfillments.all()
+    )
```

Comment:
It seems like it could generate problems with efficiency, as we need to filter every fulfillment line, get its `order_line` and the `variant_id`. @tomaszszymanski129 what do you think?

Reply:
I think this can be done simpler. I see that in function where you call this function, you're fetching fullfilments which are used here. 
Can't we do something like this:
```
f_lines = FulfillmentLine.objects.prefetch_related("order_line").filter(fulfillment__order_id=20, fulfillment__status__in=["refund", "refunded_and_returned"], order_line__variant_id__isnull=False)
```
Then you can take variant_id and the line's quantity.